### ISSUE TITLE
Feature request: register schema from URI

### ISSUE THREAD

**Issue by kool79** [harrel56/json-schema#70]
Currently library loads (registers) schema only from jsonNode/string/schemaProviderNode.
At the same time it perfectly loads depended schemas from external resources with help of schemaResolver.

The goal of this request is to provide possibility to load schema directly from URI :
```java
{
var schemaUri = new File("./mySchema.json").toUri();  // generates the "file:/" uri
var validator = new ValidatorFactory()
                          .withSchemaResolver(localFilesystemResolver)
                          .createValidator();
validator.registerSchema(schemaUri);  // new overloaded method
}

SchemaResolver localFilesystemResolver = (String uri) -> {
    try{
        if (uri.startsWith(file:/))                       // resolves the "file:/" uri
            return SchemaResolver.Result.fromProviderNode(myObjectMapper.readTree(new File()));
        return SchemaResolver.Result.empty();
    }catch (IOException e){ throw new RuntimeException(e);}
}
```

**Comment by harrel56** [comment]
If I understand correctly this would be somewhat solved with #65 ? The main difference would be that schemas will be loaded & parsed only when explicitly needed

**Comment by kool79** [comment]
Exactly )
I saw this issue but at that moment I did not understand the purpose .
Closing this as duplicate.
